packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx

```diff
@@ -295,7 +295,7 @@ const VisitTable: React.FC<VisitTableProps> = ({ showAllEncounters, visits, pati
                                   className={styles.menuItem}
                                   itemText={t('deleteThisEncounter', 'Delete this encounter')}
                                   onClick={() => {
-                                    handleDeleteEncounter(visits[index]?.id, visits[index]?.form?.display);
+                                    handleDeleteEncounter(paginatedVisits[index]?.id, paginatedVisits[index]?.form?.display);
```

Comment:
The problem here is that, fundamentally, the `index` cannot be correctly used to index into the field. This is because on each page, the index gets reset. We'd actually have to do something more similar to what was done [here](https://github.com/openmrs/openmrs-esm-patient-management/commit/8269de2166c5b72cb592abd2fb6e50ec1f21c3ed#diff-91697200229a0a527eef8c84127f7a7705b98c7e27235d547b03eaf0183d7fcaR248).

Reply:
This has been addressed appropriately in the latest commit.